### PR TITLE
RFC for Ruby dependency changes

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

 [Readable](https://github.com/paketo-buildpacks/rfcs/blob/ruby-dependencies/text/ruby/0000-ruby-dependencies.md)
 
Resolves step 1 of the following issues:
- https://github.com/paketo-buildpacks/bundler/issues/386
- https://github.com/paketo-buildpacks/mri/issues/393
- https://github.com/paketo-buildpacks/passenger/issues/254

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: text/ruby/0000-ruby-dependencies.md
Comment (sophiewigmore) [reply text/ruby/0000-ruby-dependencies.md]: Good idea, I've added that

**Review by ryanmoran — APPROVED** [review]

**Comment by ryanmoran** [comment]
WG 2022-09-27 Notes:
@sophiewigmore gave some background context on the RFC. Waiting for some more feedback from other @paketo-buildpacks/ruby-maintainers.

File: text/ruby/0000-ruby-dependencies.md
Comment (robdimsdale) [inline text/ruby/0000-ruby-dependencies.md:55]: Can we expand on what we mean by "not provided by trusted sources"? I assume this means a couple of things:
- the author of `curl` doesn't provide pre-built binaries hosted on the official site with checksums
- the statically-linked `curl` is built and distributed by a third-party (i.e. https://github.com/moparisthebest/static-curl) which we don't have sufficient confidence in.

I like the idea of downloading a single statically-linked `curl` for all x86_64 stacks, but I could see arguments against relying on a third-party github repo to obtain a pre-built version of that.

File: text/ruby/0000-ruby-dependencies.md
Comment (robdimsdale) [inline text/ruby/0000-ruby-dependencies.md:84]: While I think we could leverage `apt` to install curl and then copy the relevant files, I think it's probably about as complicated as installing from scratch ourselves, if not more complex.

**Review by robdimsdale — APPROVED** [review]
I have a couple of clarifying questions and comments but they are not blockers. Approving.

File: text/ruby/0000-ruby-dependencies.md
Comment (sophiewigmore) [reply text/ruby/0000-ruby-dependencies.md:55]: precisely all options are provided by third parties which I think is less than ideal

Comment (sophiewigmore) [reply text/ruby/0000-ruby-dependencies.md]: fixed.

**Comment by sophiewigmore** [comment]
I believe this should be numbered 0005, the other Ruby RFCs are located inside an extra rfcs subdirectory. I can move them up to the same level as this RFC in a separate PR.

File: text/ruby/0000-ruby-dependencies.md
Comment (robdimsdale) [inline text/ruby/0000-ruby-dependencies.md:55]: I don't have edit permissions, so I can't fix this myself, but

"that don't have enough confidence in to wnat to" -> "that we don't have enough confidence in to want to"

**Review by robdimsdale — APPROVED** [review]

**Pull request merged**
